python/ray/data/impl/sort.py

```diff
@@ -103,6 +103,10 @@ def sort_impl(
         result = sort_block.remote(block, boundaries, key, descending)
         map_results[i, :] = result[:-1]
         map_meta.append(result[-1])
+
+    # Early release memory.
+    del blocks
+
```

Comment:
I was looking at our sort last night and noticed that we weren't releasing the input blocks after starting the map _and_ the map results after starting the reduce/merge. It'd be nice if we could release the latter as well down below:
```python
del map_results
merge_bar = ProgressBar("Sort Merge", len(reduce_results))
merge_bar.block_until_complete([ret[0] for ret in reduce_results])
```

This can wait for a follow-up PR, if you'd like.

Reply:
Good point. I'll make a mini follow-up including this.